synapse/storage/data_stores/main/devices.py

```diff
@@ -799,6 +809,20 @@ def _txn(txn):
 
         return rows
 
+    async def _drop_device_lists_outbound_last_success_non_unique_idx(
+        self, progress, batch_size
+    ):
+        def f(conn):
+            txn = conn.cursor()
+            txn.execute("DROP INDEX IF EXISTS device_lists_outbound_last_success_idx")
+            txn.close()
```

Comment:
Isn't this function missing a `conn.commit()`? Also, why not run it with `runInteraction` so we just get a cursor and don't need to care about the connection?

Reply:
I dunno. I just cargo-culted this from similar code elsewhere.

I *think* it's something to do with schema-changing commands having weird interactions with cursors/transactions. but I really don't know.

Reply:
right, I did some digging. It seems to have originated with https://github.com/matrix-org/synapse/commit/5a32e9273ec9759caf09d5b8204dd29e7a007b97 - or rather a predecessor to that change. As far as I can see there is no reason for it to still be done this way, so I've changed it.